src/util/metadata/side_metadata/global.rs

```diff
@@ -97,6 +99,515 @@ impl SideMetadataSpec {
+        #[cfg(not(feature = "global_alloc_bit"))]
+        debug_assert!(start.is_aligned_to(BYTES_IN_PAGE) && meta_byte_lshift(self, start) == 0);
+
+        #[cfg(feature = "extreme_assertions")]
+        sanity::verify_bzero(self, start, size);
+
+        let meta_start = address_to_meta_address(self, start);
+        if cfg!(target_pointer_width = "64") || self.is_global {
+            memory::zero(
+                meta_start,
+                address_to_meta_address(self, start + size) - meta_start,
+            );
+        }
+        #[cfg(target_pointer_width = "32")]
+        if !self.is_global {
+            // per chunk policy-specific metadata for 32-bits targets
+            let chunk_num = ((start + size).align_down(BYTES_IN_CHUNK)
+                - start.align_down(BYTES_IN_CHUNK))
+                / BYTES_IN_CHUNK;
+            if chunk_num == 0 {
+                memory::zero(
+                    meta_start,
+                    address_to_meta_address(self, start + size) - meta_start,
+                );
+            } else {
+                let second_data_chunk = start.align_up(BYTES_IN_CHUNK);
+                // bzero the first sub-chunk
+                memory::zero(
+                    meta_start,
+                    address_to_meta_address(self, second_data_chunk) - meta_start,
+                );
+                let last_data_chunk = (start + size).align_down(BYTES_IN_CHUNK);
+                let last_meta_chunk = address_to_meta_address(self, last_data_chunk);
+                // bzero the last sub-chunk
+                memory::zero(
+                    last_meta_chunk,
+                    address_to_meta_address(self, start + size) - last_meta_chunk,
+                );
+                let mut next_data_chunk = second_data_chunk;
+                // bzero all chunks in the middle
+                while next_data_chunk != last_data_chunk {
+                    memory::zero(
+                        address_to_meta_address(self, next_data_chunk),
+                        metadata_bytes_per_chunk(self.log_bytes_in_region, self.log_num_of_bits),
+                    );
+                    next_data_chunk += BYTES_IN_CHUNK;
+                }
+            }
+        }
+    }
+
+    /// This is a wrapper method for implementing side metadata access. It does nothing other than
+    /// calling the access function with no overhead, but in debug builds,
+    /// it includes multiple checks to make sure the access is sane.
+    /// * check whether the given value type matches the number of bits for the side metadata.
+    /// * check if the side metadata memory is mapped.
+    /// * check if the side metadata content is correct based on a sanity map (only for extreme assertions).
+    #[inline(always)]
+    #[allow(unused_variables)] // data_addr is not used in release build
+    fn side_metadata_access<T: MetadataValue, R: Copy, F: FnMut() -> R, V: FnMut(R)>(
```

Comment:
```suggestion
    fn side_metadata_access<T: MetadataValue, R: Copy, F: FnOnce() -> R, V: FnOnce(R)>(
```
Both `access_func` and `verify_func` are called only once, so it is OK to use `FnOnce` which has less restrictions.

Reply:
Right. I will change this.